packages/flutter/lib/src/cupertino/menu_anchor.dart

```diff
@@ -0,0 +1,3136 @@
+      onSwipeEnd: _handleSwipeEnd,
+      onSwipeCanceled: _handleSwipeCancel,
+    );
+  }
+
+  void _disposeInactiveRecognizer() {
+    if (!isSwiping && _recognizer != null) {
+      _recognizer!.dispose();
+      _recognizer = null;
+    }
+  }
+
+  void _completeSwipe() {
+    _position = null;
+    widget.onDistanceChanged(0);
+    if (!mounted) {
+      // If the widget is not mounted, safely dispose of the recognizer.
+      _disposeInactiveRecognizer();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _SwipeScope(state: this, child: widget.child);
+  }
+}
+
+/// An area that can initiate swiping.
+///
+/// This widget registers with the nearest [_SwipeRegion] and exposes its position
+/// as a [ui.Rect]. This [_SwipeSurface] will route [PointerDownEvent]s to its
+/// [_SwipeRegion]. If a routed [PointerDownEvent] results in a swipe gesture, the
+/// [_SwipeRegion] will use the combined [ui.Rect] of all registered [_SwipeSurface]s
+/// to calculate the swiping distance.
+class _SwipeSurface extends SingleChildRenderObjectWidget {
+  /// Creates a swipe surface that registers with a parent [_SwipeRegion].
+  const _SwipeSurface({required super.child, this.delay = Duration.zero, this.onStart});
+
+  /// The delay before recognizing a swipe gesture.
+  final Duration delay;
+  final VoidCallback? onStart;
+
+  @override
+  _RenderSwipeSurface createRenderObject(BuildContext context) {
+    return _RenderSwipeSurface(region: _SwipeRegion.of(context)!, delay: delay, onStart: onStart);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSwipeSurface renderObject) {
+    renderObject
+      ..region = _SwipeRegion.of(context)!
+      ..delay = delay
+      ..onStart = onStart;
+  }
+}
+
+class _RenderSwipeSurface extends RenderProxyBoxWithHitTestBehavior {
+  _RenderSwipeSurface({
+    required _SwipeRegionState region,
+    required this.delay,
+    required this.onStart,
+  }) : _region = region,
+       super(behavior: HitTestBehavior.opaque) {
+    _region.attachSurface(this);
+  }
+
+  _SwipeRegionState get region => _region;
+  _SwipeRegionState _region;
+  set region(_SwipeRegionState value) {
+    if (_region != value) {
+      _region.detachSurface(this);
+      _region = value;
+      _region.attachSurface(this);
+    }
+  }
+
+  Duration delay;
+  VoidCallback? onStart;
+
+  ui.Rect computeRect() => localToGlobal(Offset.zero) & size;
+
+  @override
+  void detach() {
+    _region.detachSurface(this);
+    super.detach();
+  }
+
+  @override
+  void dispose() {
+    _region.detachSurface(this);
+    super.dispose();
+  }
+
+  @override
+  void handleEvent(PointerEvent event, BoxHitTestEntry entry) {
+    assert(debugHandleEvent(event, entry));
+    if (event is PointerDownEvent) {
+      _region.beginSwipe(event, delay: delay, onStart: onStart);
+    }
+  }
+}
+
+/// Handles swiping events for a [_SwipeRegion].
+// This class was adapted from _DragAvatar.
```

Comment:
I recommend not mentioning connections like this, because such connections can be outdated in the future as one of them is updated.